File: Francisco/francisco_prova/pw-gerenciamento-financeiro-dev/pw-gerenciamento-financeiro-dev/back-end/src/main/java/com/financeiro/backend/features/wallet/service/WalletService.java
package com.financeiro.backend.features.wallet.service;

import java.util.List;
import java.util.UUID;

import com.financeiro.backend.features.wallet.dto.request.CreateWalletRequest;
import com.financeiro.backend.features.wallet.dto.request.UpdateWalletRequest;
import com.financeiro.backend.features.wallet.dto.request.AddWalletMemberRequest;
import com.financeiro.backend.features.wallet.dto.request.UpdateWalletMemberRequest;
import com.financeiro.backend.features.wallet.dto.response.WalletMemberResponse;
import com.financeiro.backend.features.wallet.dto.response.WalletResponse;

public interface WalletService {
    WalletResponse insert(UUID ownerId, CreateWalletRequest request);
    List<WalletResponse> listByOwner(UUID ownerId);
    WalletResponse searchById(UUID id, UUID currentUserId);
    WalletResponse alter(UUID id, UUID currentUserId, UpdateWalletRequest request);
    void remove(UUID id, UUID currentUserId);
    
    void addMember(UUID walletId, UUID currentUserId, UUID targetUserId, String permission);
    List<WalletMemberResponse> listMembers(UUID walletId, UUID currentUserId);
    void addMemberByEmail(UUID walletId, UUID currentUserId, AddWalletMemberRequest request);
    void updateMember(UUID walletId, UUID currentUserId, UUID targetUserId, UpdateWalletMemberRequest request);
    void removeMember(UUID walletId, UUID currentUserId, UUID targetUserId);
}

File: Francisco/francisco_prova/pw-gerenciamento-financeiro-dev/pw-gerenciamento-financeiro-dev/Front-end/src/services/walletService.js
import api from '../config/axiosConfig';

class WalletService {
  async createWallet(data) {
    const response = await api.post(`/api/wallets`, data);
    return response;
  }

  async getWallets() {
    const response = await api.get(`/api/wallets`);
    return response;
  }

  async getWallet(id) {
    return api.get(`/api/wallets/${id}`);
  }

  async updateWallet(id, data) {
    return api.put(`/api/wallets/${id}`, data);
  }

  async deleteWallet(id) {
    return api.delete(`/api/wallets/${id}`);
  }

  async listMembers(walletId) {
    return api.get(`/api/wallets/${walletId}/members`);
  }

  async addMemberByEmail(walletId, data) {
    return api.post(`/api/wallets/${walletId}/members`, data);
  }

  async updateMember(walletId, userId, data) {
    return api.patch(`/api/wallets/${walletId}/members/${userId}`, data);
  }

  async removeMember(walletId, userId) {
    return api.delete(`/api/wallets/${walletId}/members/${userId}`);
  }

  async addMember(walletId, targetUserId, permission) {
    const response = await api.post(`/api/wallets/${walletId}/members/${targetUserId}?permission=${permission}`);
    return response;
  }
}

export default new WalletService();

